infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LogReplicationServer.java

```diff
@@ -196,10 +218,17 @@ private void handleMetadataRequest(@Nonnull RequestMsg request,
 
             LogReplicationSinkManager sinkManager = sessionToSinkManagerMap.get(session);
 
+            // SinkManager not found can indicate a new session discovered or yet to be discovered.
```

Comment:
These two snippets seem identical at L164 and L221. Would you like to have a helper method for this block?

Reply:
I tried, but the number of lines in the snippet cannot be made reduced because we want to keep the log msg her. It becomes easy to read that way.

Reply:
+1